sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,463 @@
+----
+~task_sequence();
+----
+|
+Destructor for `task_sequence`.
+
+Implicitly invokes `get` on all outstanding invocations launched through `async
+unless this `task_sequence` object was instantiated with the `balanced` property
+defined below.
+|===
+
+=== `task_sequence` Scoping
+
+`task_sequence` objects should retire all outstanding `async` invocations before
+exiting scope - this is performed by the `task_sequence` destructor unless the
+`balanced` property was specified in which case it is the programmer's
+responsibility to ensure that `get` has been called for all invocations launched
+using `async`, before the lifetime of the `task_sequence` object ends.
+
+A count of outstanding `async` invocations could be maintained by incrementing a
+private counter upon each `async` call, and decrementing it upon each `get`
+call. The destructor can call `get` on all remaining outstanding `async`
+invocations, ensuring that no `async` invocations execute beyond the object's
+lifetime.
+
+Although retiring outstanding `async` s in the destructor is necessary to ensure
+that `async` s do not outlive their `task_sequence`, it is expected that many
+common coding patterns will guarantee that the number of `async` and `get` calls
+match (are balanced) before a `task_sequence` object is destroyed, meaning that
+a simple destructor will be sufficient in these cases. To provide more
+information to the compiler and to relax the requirement for `get` to be invoked
+implicitly in the `task_sequence` destructor, the property `balanced` may be
+specified on a `task_sequence` object, which guarantees that a user will not
+allow a destructor on that `task_sequence` object to be called when there are
+outstanding `async` invocations that have not been balanced by a matching `get`
+call. In the presence of this property, potentially expensive hardware
+implementing the destructor may be elided. It is undefined behavior to specify
+the `balanced` property on `task_sequence` and then to allow the `task_sequence`
+object to be destroyed while there are any `async` invocations for which `get`
+has not been called.
+
+== `task_sequence` Properties
+
+The following code and table describe the properties that can be provided when
+declaring a `task_sequence` object.
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+struct balanced_key {
+  using value_t = property_value<balanced_key>;
+};
+
+struct invocation_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<invocation_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+struct response_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<response_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+inline constexpr balanced_key::value_t balanced;
+
+template <> struct is_property_key<balanced_key> : std::true_type {};
+
+// TODO: Not sure if the template here should just be
+// template<auto &f, class propertiesT>
```

Comment:
I think this confusion stems from the fast that an instantiation of the primary template that doesn't match the partial specialization is still well-formed.  I don't think we should allow that.

Reply:
Corrected.